dashboard/api.py

```diff
@@ -503,8 +503,9 @@ def get_certificate_url(mmtrack, course):
     if best_grade:
         course_key = best_grade.course_run.edx_course_key
         if mmtrack.financial_aid_available:
-            if best_grade.has_certificate and course.signatories.exists():
-                url = reverse('certificate', args=[best_grade.certificate.hash])
+            certificate = MicromastersCourseCertificate.objects.filter(user=mmtrack.user, course=course).first()
+            if certificate and course.signatories.exists():
+                url = reverse('certificate', args=[certificate.hash])
```

Comment:
`MMTrack` implements `get_program_certificate_url` and `program_certificate_qset`. Unless you think that's a bad approach in general, it definitely seems like we should implement `get_course_certificate_url` and `course_certificate_qset`, then call `get_course_certificate_url` from here